cppcache/src/TcrConnection.cpp

```diff
@@ -380,33 +386,33 @@ bool TcrConnection::initTcrConnection(
     }
 
     switch (acceptanceCode[0]) {
-      case REPLY_OK:
-      case SUCCESSFUL_SERVER_TO_CLIENT:
+      case static_cast<int>(acceptance_codes::REPLY_OK):
```

Comment:
If we have to static cast this switch case values than either `acceptanceCode[0]` is the wrong type or all the values of `acceptanceType[0]` are not covered by a single `enum` and the use of `enum` probably isn't appropriate.

Reply:
Oh right, we could just cast the byte we read rather than cast in every case statement.  That's way better.

Reply:
At the very least, an enum class doesn't seem appropriate.

Reply:
Usage of `enum class` should be preferred to scope the constants and avoid collision with others not to mention avoid the strangely behaving type decaying old school `enum` does.

Reply:
Exactly, it's good already, just cast in one place instead of 5 (6?).

Reply:
Jesus, I'm getting rusty...